File: app/src/main/java/org/dhis2/utils/optionset/OptionSetViewHolder.java
package org.dhis2.utils.optionset;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import org.dhis2.databinding.ItemOptionBinding;
import org.dhis2.utils.custom_views.OptionSetOnClickListener;
import org.hisp.dhis.android.core.option.Option;

public class OptionSetViewHolder extends RecyclerView.ViewHolder {

    private ItemOptionBinding binding;

    OptionSetViewHolder(@NonNull ItemOptionBinding itemView) {
        super(itemView.getRoot());
        this.binding = itemView;
    }

    public void bind(Option option, OptionSetOnClickListener listener) {
        binding.setOption(option.displayName());
        binding.executePendingBindings();

        itemView.setOnClickListener(view -> listener.onSelectOption(option));
    }
}

File: app/src/main/java/org/dhis2/utils/optionset/OptionSetViewHolder.kt
package org.dhis2.utils.optionset

import androidx.recyclerview.widget.RecyclerView

import org.dhis2.databinding.ItemOptionBinding
import org.dhis2.utils.custom_views.OptionSetOnClickListener
import org.hisp.dhis.android.core.option.Option

class OptionSetViewHolder internal constructor(private val binding: ItemOptionBinding) : RecyclerView.ViewHolder(binding.root) {

    fun bind(option: Option, listener: OptionSetOnClickListener) {
        binding.option = option.displayName()
        binding.executePendingBindings()

        itemView.setOnClickListener { listener.onSelectOption(option) }
    }
}
